tests/plugins/format/plugin.js

```diff
@@ -25,5 +25,24 @@ bender.test( {
 		bot.setHtmlWithSelection( '<fieldset><legend>^foo</legend><form>bar</form></fieldset>' );
 		var name = 'Format', combo = ed.ui.get( name );
 		assert.areSame( CKEDITOR.TRISTATE_DISABLED, combo._.state, 'check state disabled when not in context' );
+	},
+
+	// Drop-down format menu should not be toggleable (#584).
+	'test apply format style to preformated text': function() {
+		var bot = this.editorBot,
+			editor = this.editor,
+			name = 'Format',
+			combo = editor.ui.get( name );
+
+		bot.setHtmlWithSelection( '<h1>f^oo</h1>' );
+
+		// Apply the same style format to element already styled.
+		assert.areSame( CKEDITOR.TRISTATE_OFF, combo.getState(), 'check state OFF' );
+		bot.combo( name, function( combo ) {
+			assert.areSame( CKEDITOR.TRISTATE_ON, combo.getState(), 'check state ON when opened' );
+			combo.onClick( 'h1' );
+			assert.areSame( '<h1>f^oo</h1>', bot.htmlWithSelection(), 'applied h1 block style' );
```

Comment:
> applied h1 block style

`h1` was not applied here as it was already there. The test (and assert) check if `h1` is not removed when applied second time, so maybe _preserved h1 block style_ or something similar?